device/core/test/_module_client_test.js

```diff
@@ -152,4 +183,117 @@ describe('ModuleClient', function() {
+      it('starts listening for messages when a listener subscribes to the message event', function () {
+        var dummyTransport = new FakeTransport();
+        sinon.spy(dummyTransport, testConfig.enableFunc);
+        var client = new ModuleClient(dummyTransport);
+
+        // Calling 'on' twice to make sure it's called only once on the receiver.
+        // It works because the test will fail if the test callback is called multiple times, and it's called for every time the testConfig.eventName event is subscribed on the receiver.
+        client.on(testConfig.eventName, function () { });
+        client.on(testConfig.eventName, function () { });
+        assert.isTrue(dummyTransport[testConfig.enableFunc].calledOnce);
+      });
+
+      /*Tests_SRS_NODE_INTERNAL_CLIENT_18_015: [ The client shall stop listening for messages from the service whenever the last listener unsubscribes from the `inputMessage` event. ]*/
+      it('stops listening for messages when the last listener has unsubscribed', function (testCallback) {
+        var dummyTransport = new FakeTransport();
+        sinon.spy(dummyTransport, testConfig.enableFunc);
+        sinon.spy(dummyTransport, testConfig.disableFunc);
+        sinon.spy(dummyTransport, 'removeAllListeners');
+
+        var client = new ModuleClient(dummyTransport);
+        var listener1 = function () { };
+        var listener2 = function () { };
+        client.on(testConfig.eventName, listener1);
+        client.on(testConfig.eventName, listener2);
+
+        process.nextTick(function() {
+          client.removeListener(testConfig.eventName, listener1);
+          assert.isTrue(dummyTransport[testConfig.disableFunc].notCalled);
+          client.removeListener(testConfig.eventName, listener2);
+          assert(dummyTransport[testConfig.disableFunc].calledOnce);
+          testCallback();
+        });
+      });
+
+      /*Tests_SRS_NODE_INTERNAL_CLIENT_18_017: [ The client shall emit an `error` if connecting the transport fails while subscribing to `inputMessage` events. ]*/
+      it('emits an error if it fails to start listening for messages', function (testCallback) {
+        var dummyTransport = new FakeTransport();
+        var fakeError = new Error('fake');
+        sinon.stub(dummyTransport, testConfig.enableFunc).callsFake(function (callback) { callback(fakeError); });
+        var client = new ModuleClient(dummyTransport);
+        client.on('error', function (err) {
+          assert.strictEqual(err, fakeError);
+          testCallback();
+        })
+
+        // Calling 'on' twice to make sure it's called only once on the receiver.
+        // It works because the test will fail if the test callback is called multiple times, and it's called for every time the testConfig.eventName event is subscribed on the receiver.
```

Comment:
remove these 2 comments please.  It looks like copypasta from the 18_014 test above.